src/VGriveClient.vala

```diff
@@ -191,21 +191,22 @@ namespace App {
             // Check if we have changes in files and sync them
             if (this.is_syncing ()) {
                 this.log_level=0;
-                this.check_deleted_files ();
                 this.check_remote_files (this.main_path);
                 this.check_local_files (this.main_path);
+                this.check_deleted_files ();
```

Comment:
As I said in https://github.com/bcedu/VGrive/issues/70 , the `check_deleted_files` must be the first method called:

> The method check_deleted_files it's always the first method called. I need to process the deleted files at the begining because this method is the one which deketes the files from the VGrive library, so when 'check_remote_files' or check_local_files
are called I'm sure that the files that exist on the computer or on Google Drive are the ones that have to be and must be uploaded/downloaded.

Is the user deletes a local file and the method `check_remote_files` is called before `check_deleted_files`, VGrive will detect a file that exist on Google Drive but doesn't exist in local so it will be downloaded.

Reply:
Ok, I got it.
Yesterday night I was thinking about it but couldn't undo in time.